`make docs` fails

```
$ make docs 
build-tools/cli-help-generator.py CLI.in.md CLI.md
Traceback (most recent call last):
  File "/home/serhiy/neuromation/platform-client-python/build-tools/cli-help-generator.py", line 210, in <module>
    main()
  File "/home/serhiy/neuromation/platform-client-python/build-tools/cli-help-generator.py", line 203, in main
    info = parse_command(None, cli, ["neuro"])
  File "/home/serhiy/neuromation/platform-client-python/build-tools/cli-help-generator.py", line 83, in parse_command
    parse_command(ctx, sub_cmd, stack + [command_name])
  File "/home/serhiy/neuromation/platform-client-python/build-tools/cli-help-generator.py", line 73, in parse_command
    info = parse_doc(ctx, command, stack)
  File "/home/serhiy/neuromation/platform-client-python/build-tools/cli-help-generator.py", line 44, in parse_doc
    command.format_help_text(ctx, formatter)
  File "/home/serhiy/neuromation/platform-client-python/neuro-cli/src/neuro_cli/utils.py", line 211, in format_help_text
    format_example(example, formatter)
  File "/home/serhiy/neuromation/platform-client-python/neuro-cli/src/neuro_cli/utils.py", line 152, in format_example
    formatter.write_text("\b\n" + " ".join(shlex.split(line)))
  File "/home/serhiy/.pyenv/versions/3.10.4/lib/python3.10/shlex.py", line 315, in split
    return list(lex)
  File "/home/serhiy/.pyenv/versions/3.10.4/lib/python3.10/shlex.py", line 300, in __next__
    token = self.get_token()
  File "/home/serhiy/.pyenv/versions/3.10.4/lib/python3.10/shlex.py", line 109, in get_token
    raw = self.read_token()
  File "/home/serhiy/.pyenv/versions/3.10.4/lib/python3.10/shlex.py", line 191, in read_token
    raise ValueError("No closing quotation")
ValueError: No closing quotation
make: *** [Makefile:115: docs] Помилка 1
```

Python 3.10.4, the newly created virtual environment.

--- a/CLI.md
+++ b/CLI.md
@@ -2517,7 +2517,7 @@ Name | Description|
 |_\--pass-config / --no-pass-config_|Upload neuro config to the job  \[default: no\-pass-config]|
 |_\--port-forward LOCAL\_PORT:REMOTE_RORT_|Forward port\(s) of a running job to local port\(s) \(use multiple times for forwarding several ports)|
 |_\-s, --preset PRESET_|Predefined resource configuration \(to see available values, run `neuro config show`)|
-|_--privileged_|Run job in privileged mode, if it is supported by cluster.  \[default: False]|
+|_--privileged_|Run job in privileged mode, if it is supported by cluster.|
 |_\--restart \[never &#124; on-failure &#124; always]_|Restart policy to apply when a job exits  \[default: never]|
 |_\--schedule-timeout TIMEDELTA_|Optional job schedule timeout in the format '3m4s' \(some parts may be missing).|
 |_--share USER_|Share job write permissions to user or role.|
@@ -2851,7 +2851,7 @@ Name | Description|
 |Usage|Description|
 |---|---|
 | _[neuro storage cp](#neuro-storage-cp)_| Copy files and directories |
-| _[neuro storage df](#neuro-storage-df)_| Show current storage usage or storage usage of which path is a part |
+| _[neuro storage df](#neuro-storage-df)_| Show current storage usage |
 | _[neuro storage glob](#neuro-storage-glob)_| List resources that match PATTERNS |
 | _[neuro storage ls](#neuro-storage-ls)_| List directory contents |
 | _[neuro storage mkdir](#neuro-storage-mkdir)_| Make directories |
@@ -2923,7 +2923,7 @@ Name | Description|
 
 ### neuro storage df
 
-Show current storage usage or storage usage of which path is a part.
+Show current storage usage.<br/><br/>If PATH is specified, show storage usage of which path is a part.
 
 **Usage:**
 
@@ -3650,7 +3650,7 @@ Name | Description|
 |_\--pass-config / --no-pass-config_|Upload neuro config to the job  \[default: no\-pass-config]|
 |_\--port-forward LOCAL\_PORT:REMOTE_RORT_|Forward port\(s) of a running job to local port\(s) \(use multiple times for forwarding several ports)|
 |_\-s, --preset PRESET_|Predefined resource configuration \(to see available values, run `neuro config show`)|
-|_--privileged_|Run job in privileged mode, if it is supported by cluster.  \[default: False]|
+|_--privileged_|Run job in privileged mode, if it is supported by cluster.|
 |_\--restart \[never &#124; on-failure &#124; always]_|Restart policy to apply when a job exits  \[default: never]|
 |_\--schedule-timeout TIMEDELTA_|Optional job schedule timeout in the format '3m4s' \(some parts may be missing).|
 |_--share USER_|Share job write permissions to user or role.|

--- a/build-tools/site-help-generator.py
+++ b/build-tools/site-help-generator.py
@@ -3,6 +3,7 @@
 import re
 import shlex
 import sys
+import textwrap
 from pathlib import Path
 
 import click
@@ -13,6 +14,10 @@ from neuro_cli.topics import topics
 from neuro_cli.utils import split_examples
 
 HERE = Path(sys.argv[0]).resolve().parent
+
+
+def get_help(cmd):
+    return textwrap.dedent(cmd.help).strip()
 
 
 def gen_command(out, cmd, parent_ctx):
@@ -33,7 +38,7 @@ def gen_command(out, cmd, parent_ctx):
         out.append(f"{ctx.command_path} " + " ".join(pieces))
         out.append("```\n")
 
-        help, *examples = split_examples(cmd.help)
+        help, *examples = split_examples(get_help(cmd))
         help2 = click.unstyle(help)
         help3 = re.sub(r"([A-Z0-9\-]{3,60})", r"`\1`", help2)
         out.append(wrap_text(help3))
@@ -141,7 +146,7 @@ def gen_group(group, target_path, parent_ctx):
         out.append("```")
         out.append("")
 
-        out.append(click.unstyle(group.help))
+        out.append(click.unstyle(get_help(group)))
         out.append("")
 
         commands = []
@@ -198,7 +203,7 @@ def gen_shortcuts(commands, target_path, ctx):
 def gen_topics(target_path, ctx):
     for name in topics.list_commands(ctx):
         topic = topics.get_command(ctx, name)
-        out = [topic.help]
+        out = [get_help(topic)]
         fname = target_path / f"topic-{topic.name}.md"
         fname.write_text("\n".join(out))
 

--- a/neuro-cli/docs/job.md
+++ b/neuro-cli/docs/job.md
@@ -344,7 +344,7 @@ $ neuro run -s cpu-small --entrypoint=/script.sh image:my-ubuntu:latest -- arg1 
 | _--pass-config / --no-pass-config_ | Upload neuro config to the job  _\[default: no-pass-config\]_ |
 | _--port-forward LOCAL\_PORT:REMOTE\_RORT_ | Forward port\(s\) of a running job to local port\(s\) \(use multiple times for forwarding several ports\) |
 | _-s, --preset PRESET_ | Predefined resource configuration \(to see available values, run `neuro config show`\) |
-| _--privileged_ | Run job in privileged mode, if it is supported by cluster.  _\[default: False\]_ |
+| _--privileged_ | Run job in privileged mode, if it is supported by cluster. |
 | _--restart \[never &#124; on-failure &#124; always\]_ | Restart policy to apply when a job exits  _\[default: never\]_ |
 | _--schedule-timeout TIMEDELTA_ | Optional job schedule timeout in the format '3m4s' \(some parts may be missing\). |
 | _--share USER_ | Share job write permissions to user or role. |

--- a/neuro-cli/docs/shortcuts.md
+++ b/neuro-cli/docs/shortcuts.md
@@ -629,7 +629,7 @@ $ neuro run -s cpu-small --entrypoint=/script.sh image:my-ubuntu:latest -- arg1 
 | _--pass-config / --no-pass-config_ | Upload neuro config to the job  _\[default: no-pass-config\]_ |
 | _--port-forward LOCAL\_PORT:REMOTE\_RORT_ | Forward port\(s\) of a running job to local port\(s\) \(use multiple times for forwarding several ports\) |
 | _-s, --preset PRESET_ | Predefined resource configuration \(to see available values, run `neuro config show`\) |
-| _--privileged_ | Run job in privileged mode, if it is supported by cluster.  _\[default: False\]_ |
+| _--privileged_ | Run job in privileged mode, if it is supported by cluster. |
 | _--restart \[never &#124; on-failure &#124; always\]_ | Restart policy to apply when a job exits  _\[default: never\]_ |
 | _--schedule-timeout TIMEDELTA_ | Optional job schedule timeout in the format '3m4s' \(some parts may be missing\). |
 | _--share USER_ | Share job write permissions to user or role. |

--- a/neuro-cli/docs/storage.md
+++ b/neuro-cli/docs/storage.md
@@ -14,7 +14,7 @@ Storage operations.
 | Usage | Description |
 | :--- | :--- |
 | [_cp_](storage.md#cp) | Copy files and directories |
-| [_df_](storage.md#df) | Show current storage usage or storage... |
+| [_df_](storage.md#df) | Show current storage usage |
 | [_glob_](storage.md#glob) | List resources that match PATTERNS |
 | [_ls_](storage.md#ls) | List directory contents |
 | [_mkdir_](storage.md#mkdir) | Make directories |
@@ -106,7 +106,7 @@ $ neuro cp storage:results/*.out .
 
 ### df
 
-Show current storage usage or storage...
+Show current storage usage
 
 
 #### Usage
@@ -115,7 +115,10 @@ Show current storage usage or storage...
 neuro storage df [OPTIONS] [PATH]
 ```
 
-Show current storage usage or storage usage of which path is a part.
+Show current storage usage.
+
+If `PATH` is specified, show storage usage of
+which path is a part.
 
 #### Options
 

--- a/neuro-cli/src/neuro_cli/config.py
+++ b/neuro-cli/src/neuro_cli/config.py
@@ -229,7 +229,8 @@ async def get_clusters(root: Root) -> None:
 @command()
 @argument("cluster_name", required=False, default=None, type=CLUSTER_ALLOW_UNKNOWN)
 async def switch_cluster(root: Root, cluster_name: Optional[str]) -> None:
-    """Switch the active cluster.
+    """
+    Switch the active cluster.
 
     CLUSTER_NAME is the cluster name to select.  The interactive prompt is used if the
     name is omitted (default).
@@ -255,7 +256,8 @@ async def switch_cluster(root: Root, cluster_name: Optional[str]) -> None:
 @command()
 @argument("org_name", required=True, type=ORG_ALLOW_UNKNOWN)
 async def switch_org(root: Root, org_name: Optional[str]) -> None:
-    """Switch the active organization.
+    """
+    Switch the active organization.
 
     ORG_NAME is the organization name to select. Use literal "NO_ORG" to switch
     to using current cluster directly instead of on behalf of some org.

--- a/neuro-cli/src/neuro_cli/stats.py
+++ b/neuro-cli/src/neuro_cli/stats.py
@@ -118,7 +118,7 @@ def make_record(uid: str, url: URL, cmd: str, args: str, version: str) -> str:
         "aid": str(url),               # application id, https://dev.neu.ro/api/v1
     }
     # fmt: on
-    return urlencode(ret, quote_via=urlquote)  # type: ignore
+    return urlencode(ret, quote_via=urlquote)
 
 
 async def send(client: Client, uid: str, data: List[sqlite3.Row]) -> None:

--- a/neuro-cli/src/neuro_cli/storage.py
+++ b/neuro-cli/src/neuro_cli/storage.py
@@ -229,7 +229,9 @@ async def glob(root: Root, patterns: Sequence[URL]) -> None:
 )
 async def df(root: Root, path: Optional[URL]) -> None:
     """
-    Show current storage usage or storage usage of which path is a part.
+    Show current storage usage.
+
+    If PATH is specified, show storage usage of which path is a part.
     """
     usage = await root.client.storage.disk_usage(uri=path)
     root.print(DiskUsageFormatter()(usage))
@@ -694,7 +696,8 @@ async def mv(
 async def tree(
     root: Root, path: URL, size: bool, human_readable: bool, sort: str, show_all: bool
 ) -> None:
-    """List storage in a tree-like format
+    """
+    List storage in a tree-like format
 
     Tree is a recursive directory listing program that produces a depth indented listing
     of files, which is colorized ala dircolors if the LS_COLORS environment variable is

--- a/neuro-cli/src/neuro_cli/topics.py
+++ b/neuro-cli/src/neuro_cli/topics.py
@@ -448,7 +448,8 @@ async def aliases() -> None:
 
 @topics.command()
 async def secrets() -> None:
-    """Using secrets
+    """\
+    Using secrets
     =============
 
     A *secret* is a piece of encrypted named data stored in the Neuro Platform Cluster.
@@ -496,7 +497,8 @@ async def secrets() -> None:
 
 @topics.command()
 async def sharing() -> None:
-    """Using the sharing functionality
+    """\
+    Using the sharing functionality
     ===============================
 
     Understanding permissions

--- a/neuro-cli/src/neuro_cli/utils.py
+++ b/neuro-cli/src/neuro_cli/utils.py
@@ -8,6 +8,7 @@ import re
 import shlex
 import shutil
 import sys
+import textwrap
 from datetime import timedelta
 from typing import (
     Any,
@@ -196,6 +197,7 @@ class NeuroClickMixin:
         """Writes the help text to the formatter if it exists."""
         deprecated = self.deprecated  # type: ignore
         help = self.help  # type: ignore
+        help = help and textwrap.dedent(help)
         if help:
             help_text, *examples = split_examples(help)
             if help_text:

--- a/neuro-sdk/src/neuro_sdk/_azure_bucket_provider.py
+++ b/neuro-sdk/src/neuro_sdk/_azure_bucket_provider.py
@@ -148,12 +148,14 @@ class AzureProvider(MeasureTimeDiffMixin, BucketProvider):
     ) -> None:
         blob_client = self._client.get_blob_client(key)
         if isinstance(body, bytes):
-            await blob_client.upload_blob(body)
+            # XXX (S Storchaka 2022-04-14): Incorrect annotation or bug?
+            await blob_client.upload_blob(body)  # type: ignore
         else:
             blocks = []
             async for data in body:
                 block_id = secrets.token_hex(16)
-                await blob_client.stage_block(block_id, data)
+                # XXX (S Storchaka 2022-04-14): Incorrect annotation or bug?
+                await blob_client.stage_block(block_id, data)  # type: ignore
                 if progress:
                     await progress(len(data))
                 blocks.append(BlobBlock(block_id=block_id))

